src/webgpu/capability_info.ts

```diff
@@ -109,26 +109,35 @@ export const kPerStageBindingLimits: {
   'storage-texture': 4,
 };
 
+type BindingKind =
+  | 'uniform-buffer'
+  | 'storage-buffer'
+  | 'sampler'
+  | 'comparison-sampler'
+  | 'sampled-texture'
+  | 'storage-texture';
+
 const kStagesAll = C.ShaderStage.Vertex | C.ShaderStage.Fragment | C.ShaderStage.Compute;
 const kStagesCompute = C.ShaderStage.Compute;
 export const kBindingTypeInfo: {
   [k in GPUBindingType]: {
     type: 'buffer' | 'texture' | 'sampler';
+    kind: BindingKind;
```

Comment:
I'm going to try to clean this up (and maybe merge some stuff with validation_test.ts) separately.

Reply:
#167

Reply:
Ok